documentation/specs/CodeVisualizer.md

```diff
@@ -0,0 +1,135 @@
+# `CodeVisualizer` Component Specification
+
+## Design
+
+`CodeVisualizer` wraps one or more `CodeBlock`s with a header to allow the user to switch between blocks.
+
+`CodeVisualizer` will be a controlled component. It will require the consumer to provide a selected language and manage its state.
+
+### API
+
+```ts
+export type CodeVisualizerProps = {
+  /**
+   * CodeVisualizer content. This should be a header and collection of snippets.
+   */
+  children: ReactNode;
```

Comment:
if there's only one snippet provided, the CodeVisualizer goes into "View Only" mode and doesn't show a language dropdown